src/components/pages/devices/flyouts/deviceNew/deviceNew.js

```diff
@@ -206,6 +219,16 @@ export class DeviceNew extends LinkedComponent {
     return true;
   }
 
+  fetchDeviceModelOptions = () => {
+    this.setState({ deviceModels: { isPending: true } });
+
+    this.deviceModelSubscription = DeviceSimulationService.getDeviceModelSelectOptions()
```

Comment:
I believe device models are an app level setting. They don't change much, right? They are more like config settings from the server. Loading the device models should probably be placed in the app reducers/epics.